Bio/Entrez/Parser.py

```diff
@@ -169,12 +169,19 @@ def __str__(self):
 class DataHandler(object):
 
     import platform
-    if platform.system() == 'Windows':
-        directory = os.path.join(os.getenv("APPDATA"), "biopython")
-    else:  # Unix/Linux/Mac
-        home = os.path.expanduser('~')
-        directory = os.path.join(home, '.config', 'biopython')
-        del home
+    from Bio import Entrez
+
+    # Check if user has set a custom cache location
+    if Entrez.cache is not None:
+        directory = Entrez.cache
```

Comment:
Should we add a check here? Maybe:

```python
if not os.path.isdir(directory):
    raise ValueError("Please set Entrez.cache to a directory name (or None)")
```

Reply:
Just confirming I understand, this would mean the cache must be an existing directory that user creates before running Entrez, right? In my test code, I used the cache parameter to have Entrez create the tmp folder ad hoc. If this is bad form, it makes sense to me to enforce that the cache must pre-exist.

Reply:
True, this could be done either way since the code will create the directory if it does not exist... its a design choice. Now that you've pointed this out I have no strong view either way.